lib/nokogiri/css/xpath_visitor.rb

```diff
@@ -51,7 +51,8 @@ def visit_function node
         when /^comment\(/
           "comment()"
         when /^has\(/
-          ".//#{node.value[1].accept(self)}"
+          is_direct = node.value[1].value[0].nil? # e.g. "has(> a)", "has(~ a)", "has(+ a)"
```

Comment:
Is the comment here gratuitous?  Is there a more self-documenting name for `is_direct`?

Reply:
I'm fine with both the comment and the name. No worries!